### ISSUE TITLE
CS1503	Argument 1: cannot convert from 'System.Collections.Specialized.NameValueCollection' to 'System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, Microsoft.Extensions.Primitives.StringValues>>

### ISSUE THREAD

**Issue by zfteo** [nozzlegear/ShopifySharp#162]
Hi I have an issues for IsAuthenticRequest.

The type of the input for IsAuthenticRequest doesn't match.

var qs = Request.QueryString;
if(AuthorizationService.IsAuthenticRequest(qs, shopifySecretKey))
{
    //Request is authentic.
}
else
{
    //Request is not authentic and should not be acted on.
}

**Comment by nozzlegear** [comment]
Hey thanks for reporting this! When we updated the package to v4 with .NET Core/Standard support I wasn't able to use the old `NameValueCollection` type for these authorization methods anymore. You'll need to convert the `Request.QueryString` to a list of KVPs. I wrote a blog post here that covers doing that, which should get this working again: https://nozzlegear.com/blog/shopifysharp-version-4-migration-guide
